File: spec/factories/revisions.rb
# == Schema Information
#
# Table name: revisions
#
#  id         :integer          not null, primary key
#  post_id    :integer
#  title      :text
#  summary    :text
#  goal       :text
#  event_url  :text
#  body       :text
#  slide_url  :text
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  comment    :text
#

FactoryBot.define do
  factory :revision do
    title     { Faker::Lorem.characters(8) }
    summary   { Faker::Lorem.characters(8) }
    goal      { Faker::Lorem.characters(8) }
    comment   { Faker::Lorem.characters(8) }
    event_url { Faker::Lorem.characters(8) }
    body      { Faker::Lorem.characters(8) }
    slide_url { Faker::Lorem.characters(8) }
  end
end

File: frontend/src/api/__test__/factories/current_user/all/revisions.js
var faker = require('faker')
faker.locale = 'ja'

export default {
  revision1: {
    id: 1,
    post_id: 1,
    title:      faker.lorem.sentence(),
    summary:    faker.lorem.text(),
    goal:       faker.lorem.text(),
    comment:    faker.lorem.text(),
    event_url:  faker.internet.url(),
    body:       faker.lorem.text(),
    slide_url:  faker.internet.url(),
    created_at: faker.date.recent(),
    updated_at: faker.date.recent()
  },
  revision2: {
    id: 2,
    post_id: 2,
    title:      faker.lorem.sentence(),
    summary:    faker.lorem.text(),
    goal:       faker.lorem.text(),
    comment:    faker.lorem.text(),
    event_url:  faker.internet.url(),
    body:       faker.lorem.text(),
    slide_url:  faker.internet.url(),
    created_at: faker.date.recent(),
    updated_at: faker.date.recent()
  }
}
